src/dbnode/generated/proto/namespace/schema.proto

```diff
@@ -0,0 +1,29 @@
+syntax = "proto3";
+package namespace;
+
+// SchemaMeta contains schema metadata.
+message SchemaMeta {
+    // message name as it appears in proto file descriptor.
+    string messageName = 1;
+}
+
+// SchemaOptions contains schema options for a namespace.
+message SchemaOptions {
+    // repo contains a history of deployed schema defintions in form of proto file descriptors.
+    FileDescriptorRepo repo = 1;
+    // schemas contains a set of named schemas for a namespace.
+    // a namespace can have multiple schemas.
+    // series identifies its schema by schema name (the map key) and a version (repo version).
+    map<string, SchemaMeta> schemas = 2;
```

Comment:
So we'll have a history of all the binary descriptors, but only the latest "set" of message names? Does it make sense to track the history of both together or do you think we don't need that?

Also, how come we're tracking the map separately from the descriptors? Seems like you could generate a list of all the messages from the descriptors itself. Is it so we could add additional metadata thats not included in the proto syntax? That makes sense to me I think

Reply:
yes, it is for additional metadata that is on the series level (not namespace), similar to cstar table level metadata (properties), such as ttl, cp strategy, cache strategy, etc.

Reply:
> So we'll have a history of all the binary descriptors, but only the latest "set" of message names? Does it make sense to track the history of both together or do you think we don't need that?
> 
> Also, how come we're tracking the map separately from the descriptors? Seems like you could generate a list of all the messages from the descriptors itself. Is it so we could add additional metadata thats not included in the proto syntax? That makes sense to me I think

map (schemas) is not used to map proto message name to proto message descriptor.  it is used to map schema id(name) to proto message name.

Map is append only, we don't delete schema id or reuse it (to map to another message descriptor). this can be enforced in schema cli.

When client writes series, it identifies series schema via schema id.